File: Easy/0069_sqrtx/sqrtx.py
#!/bin/python3

from typing import List

class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0
        r = x
        while r * r > x:
            r = (r + x // r) // 2
        return r

def main():
    tests: List[int] = [
        4,
        8,
        16,
        0,
    ]

    for x in tests:
        print(f"---\nx= {x} => Sqrt= {Solution().mySqrt(x)}")

if __name__ == "__main__":
    main()

File: Easy/0069_sqrtx/sqrtx.go
package main

import (
	"fmt"
)

func mySqrt(x int) int {
	if (x < 2)	{
		return x
	}

	r := x
	for r * r > x {
		r = (r + x / r) / 2
	}
  return r
}

func main() {
	tests := []int{
		4,
		8,
		16,
		0,
		2,
		1,
	}

	for _, x := range tests {
		fmt.Printf("---\nx=%d => srqt= %d\n", x, mySqrt(x))
	}
}
